packages/terra-application-utility/src/utility/_UtilityMenu.jsx

```diff
@@ -0,0 +1,319 @@
+  }
+
+  buildItem(key, leftInset, rightInset) {
+    const item = this.getItem(key);
+    const chevron = hasChevron(item);
+    return (
+      <MenuItem
+        key={key}
+        itemKey={key}
+        id={item.id}
+        title={item.title}
+        content={item.content}
+        contentLocation={item.contentLocation}
+        isSelected={item.isSelected}
+        isSelectable={item.isSelectable}
+        hasChevron={chevron}
+        leftInset={leftInset}
+        rightInset={rightInset}
+        onChange={this.handleOnChange}
+        variant={this.props.variant}
+      />
+    );
+  }
+
+  buildListContent(currentItem) {
+    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
+      const leftInset = this.childrenHasCheckmark(currentItem);
+      const rightInset = this.childrenHasChevron(currentItem);
+      return (
+        <ul className={cx('body')}>
+          {currentItem.childKeys.map((key) => {
+            if (this.getItem(key).contentLocation !== Utils.LOCATIONS.FOOTER) {
+              return this.buildItem(key, leftInset, rightInset);
+            }
+            return null;
+          })}
+        </ul>
+      );
+    }
+    return null;
+  }
+
+  buildFooterContent(currentItem) {
+    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
+      return currentItem.childKeys.map((key) => {
+        let item = null;
+        if (this.getItem(key).contentLocation === Utils.LOCATIONS.FOOTER) {
+          item = this.buildItem(key);
+        }
+        return item;
+      });
+    }
+    return null;
+  }
+
+  childrenHasCheckmark(item) {
+    const childrenHasCheckmark = item.childKeys.some((key) => {
+      const currentItem = this.getItem(key);
+      return currentItem.isSelectable === true && currentItem.contentLocation !== Utils.LOCATIONS.FOOTER;
+    });
+    return childrenHasCheckmark;
+  }
+
+  childrenHasChevron(item) {
+    const childrenHasChevron = item.childKeys.some((key) => {
+      const childKeys = this.getItem(key).childKeys;
+      return childKeys && childKeys.length > 0 && this.getItem(key).contentLocation !== Utils.LOCATIONS.FOOTER;
+    });
+    return childrenHasChevron;
+  }
+
+  /**
+   * Function to trigger when an item is selected.
+   * 1. Has children: navigate to the next page
+   * 2. Toggles: trigger onChange without closing the menu.
+   * 3. Endpoint: close menu and trigger onChange.
+   * @param {*} event
+   * @param {*} key
+   */
+  handleOnChange(event, key) {
+    const childKeys = this.getItem(key).childKeys;
+    const item = this.getItem(key);
+    if (typeof childKeys !== 'undefined' && childKeys.length > 0) {
+      this.setState({
+        previousKey: this.push(this.state.currentKey),
+        currentKey: key,
+      });
+    } else {
+      if (item.isSelectable !== true) {
+        this.props.onRequestClose();
+      }
+      this.props.onChange(event, key);
+    }
+  }
+
+  handleRequestBack() {
```

Comment:
Can handleRequestBack and pop be combined into a single function? I'm not seeing them used separately.

Reply:
Combined them [here](https://github.com/cerner/terra-framework/pull/103/commits/efd64c3eed5f7146aaefa0505339e34e92f75bed#diff-69f938b44661317549223f97d5ce876fR201).